src/ui/atoms/button/Button.tsx

```diff
@@ -27,24 +37,29 @@ export type TButtonProps = Readonly<{
  * Primary UI component for user interaction.
  */
 export const Button: React.FC<TButtonProps> = ({
-  primary = false,
+  variant = 'primary',
   size = 'medium',
+  backgroundColor = 'primary',
+  disabled = false,
   label,
   onClick,
   ...props
 }: TButtonProps): JSX.Element => (
   <button
-    className={classNames(styles.core, {
-      [styles.normal]: !primary,
-      [styles.primary]: primary,
-      [styles.small]: size == 'small',
-      [styles.medium]: size == 'medium',
-      [styles.large]: size == 'large'
+    className={classNames('okp4-button-main', {
+      [`variant-${variant}`]: true,
+      [`size-${size}`]: true,
+      [`background-color-${backgroundColor}`]: true,
+      disabled: !!disabled
     })}
+    disabled={disabled}
     onClick={onClick}
+    title={label}
     type="button"
     {...props}
   >
-    {label}
+    <Typography as="div" color="highlighted-text" fontSize="small" fontWeight="light" noWrap>
```

Comment:
From what I see, you don't handle child components, so we're not quite compliant with the spec discussed in [ui/issues/119](https://github.com/okp4/ui/issues/119#issuecomment-1100130307) regarding the capabilities to display icons in the button.

Reply:
@ccamel 
I updated issue to be compliant with the specification at the moment.